--- ldap.c	2010-09-02 02:06:31.604289492 -0700
+++ ldap_32.c	2010-09-02 02:19:07.775229875 -0700
@@ -119,7 +119,7 @@ int uxds_user_authz(int select, authzdat
 	if (!ldap_get_option(ld, LDAP_OPT_X_SASL_SSF, &ssf)) {
 	    val = (unsigned long) ssf;
 	}
-	fprintf(stderr, "SSF level is: %lu\n", val);
+	fprintf(stderr, "SSF level is: %i\n", val);
     }
 #endif				/* HAVE_LDAP_SASL */
     if (auth.debug) {
@@ -213,7 +213,7 @@ int uxds_acct_parse(int bindtype, authzd
 	    kuser = get_krbname(auth, 0);
 	    if (auth.debug)
 		fprintf(stderr,
-			"user account filter half returned: %s, size %lu len %lu\n",
+			"user account filter half returned: %s, size %i len %i\n",
 			kuser, sizeof(kuser), strlen(kuser));
 	    filter = strdup(center(fbuf, "uid=", kuser));
 	    center_free(fbuf);
@@ -456,7 +456,7 @@ int uxds_acct_add(usrt pxtype, struct mo
     filter = center(cbuf, center(cbuf, POSIXGROUP, mdata.group), "))");
     //center_free(cbuf);
     if (auth.debug)
-	fprintf(stderr, "filter is %s, len %lu\n", filter, strlen(filter));
+	fprintf(stderr, "filter is %s, len %i\n", filter, strlen(filter));
     if (ldap_search_ext_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0, 
                           NULL, NULL, NULL, 0, &msg)
 	!= LDAP_SUCCESS) {
@@ -490,7 +490,7 @@ int uxds_acct_add(usrt pxtype, struct mo
     vals = ldap_get_values_len(ld, entry, "description");
     if (vals[0]->bv_val != NULL) {
 	if (auth.debug)
-	    fprintf(stderr, "%s : role, len %lu\n", vals[0]->bv_val,
+	    fprintf(stderr, "%s : role, len %i\n", vals[0]->bv_val,
 		    strlen(vals[0]->bv_val));
 	role = strdup(vals[0]->bv_val);
     }
@@ -500,7 +500,7 @@ int uxds_acct_add(usrt pxtype, struct mo
     char *gidnum = NULL;
     if (vals[0]->bv_val != NULL) {
 	if (auth.debug)
-	    fprintf(stderr, "%s : gidNumber, len %lu\n", vals[0]->bv_val,
+	    fprintf(stderr, "%s : gidNumber, len %i\n", vals[0]->bv_val,
 		    strlen(vals[0]->bv_val));
 	gidnum = strdup(vals[0]->bv_val);
     }
@@ -705,7 +705,7 @@ int uxds_acct_add(usrt pxtype, struct mo
 		    mdata.user, mdata.group, mdata.uidnum, ge_cos);
 
 	if (ldap_add_ext_s(ld, user_dn, useradd, NULL, NULL) != LDAP_SUCCESS) {
-	    fprintf(stderr, "Attempted DN: %s, len %lu\n", user_dn,
+	    fprintf(stderr, "Attempted DN: %s, len %i\n", user_dn,
 		    strlen(user_dn));
             ldap_get_option(ld, LDAP_OPT_RESULT_CODE, &rc);
             fprintf(stderr, "%s: %s\n", res, ldap_err2string(rc));
@@ -864,7 +864,7 @@ int uxds_acct_del(usrt pxtype, struct mo
 	break;
     }
     if (auth.debug)
-	fprintf(stderr, "search filter used: %s, len %lu\n", filter,
+	fprintf(stderr, "search filter used: %s, len %i\n", filter,
 		strlen(filter));
     if (ldap_search_ext_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0, 
         NULL, NULL, NULL, 0, &msg) != LDAP_SUCCESS) {
@@ -973,7 +973,7 @@ int uxds_acct_mod(usrt pxtype, struct mo
     }
     center_free(fbuf);
     if (auth.debug)
-	fprintf(stderr, "filter is %s, len %lu\n", filter, strlen(filter));
+	fprintf(stderr, "filter is %s, len %i\n", filter, strlen(filter));
     if (ldap_search_ext_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0,
                           NULL, NULL, NULL, 0, &msg) != LDAP_SUCCESS) {
         ldap_get_option(ld, LDAP_OPT_RESULT_CODE, &rc);
@@ -1022,7 +1022,7 @@ int uxds_acct_mod(usrt pxtype, struct mo
 	vals = ldap_get_values_len(ld, entry, "givenName");
 	if (vals[0]->bv_val != NULL) {
 	    if (auth.debug)
-		fprintf(stderr, "%s : first name, len %lu\n", vals[0]->bv_val,
+		fprintf(stderr, "%s : first name, len %i\n", vals[0]->bv_val,
 			strlen(vals[0]->bv_val));
 	    mdata.firstname = strdup(vals[0]->bv_val);
 	    ldap_value_free_len(vals);
@@ -1032,7 +1032,7 @@ int uxds_acct_mod(usrt pxtype, struct mo
 	vals = ldap_get_values_len(ld, entry, "sn");
 	if (vals[0]->bv_val != NULL) {
 	    if (auth.debug)
-		fprintf(stderr, "%s : sn, len %lu\n", vals[0]->bv_val,
+		fprintf(stderr, "%s : sn, len %i\n", vals[0]->bv_val,
 			strlen(vals[0]->bv_val));
 	    mdata.lastname = strdup(vals[0]->bv_val);
 	    ldap_value_free_len(vals);
@@ -1041,7 +1041,7 @@ int uxds_acct_mod(usrt pxtype, struct mo
     vals = ldap_get_values_len(ld, entry, "gecos");
     if (vals[0]->bv_val != NULL) {
 	if (auth.debug)
-	    fprintf(stderr, "%s : gecos, len %lu\n", vals[0]->bv_val,
+	    fprintf(stderr, "%s : gecos, len %i\n", vals[0]->bv_val,
 		    strlen(vals[0]->bv_val));
 	old_gecos = strdup(vals[0]->bv_val);
     }
@@ -1156,7 +1156,7 @@ int uxds_acct_mod(usrt pxtype, struct mo
 	    return 1;
 	}
 	if (ldap_modify_ext_s(ld, mod_dn, usermod, NULL, NULL) != LDAP_SUCCESS) {
-	    fprintf(stdout, "Attempted DN: %s, len %lu\n", mod_dn,
+	    fprintf(stdout, "Attempted DN: %s, len %i\n", mod_dn,
 		    strlen(mod_dn));
 #ifdef TOOL_LOG
 	    log_event(mod_dn, USER, MOD, "FAILED");
