--- sudoldap.c	2008-07-30 11:54:15.000000000 -0700
+++ sudoldap_32.c	2008-07-30 11:54:15.000000000 -0700
@@ -28,7 +28,7 @@ int uxds_sudo_add(authzdata auth, struct
     }
     center_free(cbuf);
     if (auth.debug)
-	fprintf(stderr, "filter is %s, len %lu\n", filter, strlen(filter));
+	fprintf(stderr, "filter is %s, len %i\n", filter, strlen(filter));
     if (ldap_search_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0, &msg)
 	!= LDAP_SUCCESS) {
 	ldap_perror(ld, "ldap_search_s");
@@ -117,7 +117,7 @@ int uxds_sudo_add(authzdata auth, struct
 		      ",ou=sudoers,"), auth.basedn);
     fprintf(stderr, "DN is %s\n", su_dn);
     if (ldap_add_s(ld, su_dn, sudoadd) != LDAP_SUCCESS) {
-	fprintf(stderr, "Attempted DN: %s, len %lu\n", su_dn,
+	fprintf(stderr, "Attempted DN: %s, len %i\n", su_dn,
 		strlen(su_dn));
 #ifdef TOOL_LOG
 	log_event(su_dn, SUDOER, ADD, "attempt FAILED");
@@ -150,7 +150,7 @@ int uxds_sudo_del(authzdata auth, struct
     filter =
 	strdup(center(cbuf, center(cbuf, SUDOUSER, su->sudoer), "))"));
     if (auth.debug)
-	fprintf(stderr, "filter is %s, len %lu\n", filter, strlen(filter));
+	fprintf(stderr, "filter is %s, len %i\n", filter, strlen(filter));
     if (ldap_search_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0, &msg)
 	!= LDAP_SUCCESS) {
 	ldap_perror(ld, "ldap_search_s");
@@ -206,7 +206,7 @@ int uxds_sudo_mod(authzdata auth, struct
     filter =
 	strdup(center(cbuf, center(cbuf, SUDOUSER, su->sudoer), "))"));
     if (auth.debug)
-	fprintf(stderr, "filter is %s, len %lu\n", filter, strlen(filter));
+	fprintf(stderr, "filter is %s, len %i\n", filter, strlen(filter));
     if (ldap_search_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0, &msg)
 	!= LDAP_SUCCESS) {
 	ldap_perror(ld, "ldap_search_s");
@@ -293,7 +293,7 @@ int uxds_sudo_mod(authzdata auth, struct
 		      ",ou=sudoers,"), auth.basedn);
 
     if (ldap_modify_s(ld, su_dn, sudomod) != LDAP_SUCCESS) {
-	fprintf(stdout, "Attempted DN: %s, len %lu\n", su_dn,
+	fprintf(stdout, "Attempted DN: %s, len %i\n", su_dn,
 		strlen(su_dn));
 #ifdef TOOL_LOG
 	log_event(su_dn, SUDOER, MOD, "attempt FAILED");
