--- /vicepa/git/uxdstools.git/src/sudoldap.c	2010-12-29 13:00:46.468073315 -0800
+++ /vicepa/git/uxdstools.git/src/sudoldap32.c	2011-08-16 11:27:54.303713998 -0700
@@ -40,7 +40,7 @@ int uxds_sudo_add(uxds_authz_t auth, uxd
 	    return 1;
     }
     if (auth.debug)
-	fprintf(stderr, "filter is %s, len %lu\n", filter, strlen(filter));
+	fprintf(stderr, "filter is %s, len %i\n", filter, strlen(filter));
     if (ldap_search_ext_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0,
 			  NULL, NULL, NULL, 0, &msg) != LDAP_SUCCESS) {
 	ldap_get_option(ld, LDAP_OPT_RESULT_CODE, &rc);
@@ -157,7 +157,7 @@ int uxds_sudo_add(uxds_authz_t auth, uxd
 
     fprintf(stderr, "DN is %s\n", su_dn);
     if (ldap_add_ext_s(ld, su_dn, sudoadd, NULL, NULL) != LDAP_SUCCESS) {
-	fprintf(stderr, "Attempted DN: %s, len %lu\n", su_dn,
+	fprintf(stderr, "Attempted DN: %s, len %i\n", su_dn,
 		strlen(su_dn));
 #ifdef TOOL_LOG
 	log_event(su_dn, SUDOER, ADD, "attempt FAILED");
@@ -196,7 +196,7 @@ int uxds_sudo_del(uxds_authz_t auth, uxd
     if (!snprintf(filter, SU_LEN, SUDOUSER, su->sudoer))
 	return 1;
     if (auth.debug)
-	fprintf(stderr, "filter is %s, len %lu\n", filter, strlen(filter));
+	fprintf(stderr, "filter is %s, len %i\n", filter, strlen(filter));
     if (ldap_search_ext_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0,
 			  NULL, NULL, NULL, 0, &msg) != LDAP_SUCCESS) {
 	ldap_get_option(ld, LDAP_OPT_RESULT_CODE, &rc);
@@ -256,7 +256,7 @@ int uxds_sudo_mod(uxds_authz_t auth, uxd
     if (!snprintf(filter, SU_LEN, SUDOUSER, su->sudoer))
 	return 1;
     if (auth.debug)
-	fprintf(stderr, "filter is %s, len %lu\n", filter, strlen(filter));
+	fprintf(stderr, "filter is %s, len %i\n", filter, strlen(filter));
     if (ldap_search_ext_s(ld, NULL, LDAP_SCOPE_SUBTREE, filter, NULL, 0,
 			  NULL, NULL, NULL, 0, &msg) != LDAP_SUCCESS) {
 	ldap_get_option(ld, LDAP_OPT_RESULT_CODE, &rc);
@@ -346,7 +346,7 @@ int uxds_sudo_mod(uxds_authz_t auth, uxd
         return 1;
 
     if (ldap_modify_ext_s(ld, su_dn, sudomod, NULL, NULL) != LDAP_SUCCESS) {
-	fprintf(stdout, "Attempted DN: %s, len %lu\n", su_dn,
+	fprintf(stdout, "Attempted DN: %s, len %i\n", su_dn,
 		strlen(su_dn));
 #ifdef TOOL_LOG
 	log_event(su_dn, SUDOER, MOD, "attempt FAILED");
